crates/sui-source-validation/src/tests.rs

```diff
@@ -30,23 +33,85 @@ async fn successful_verification() -> anyhow::Result<()> {
         publish_package(context, sender, b_src).await
     };
 
-    let a_pkg = {
+    let (a_pkg, a_ref) = {
         let fixtures = tempfile::tempdir()?;
         let b_id = b_ref.0.into();
         copy_package(&fixtures, "b", [("b", b_id)]).await?;
         let a_src = copy_package(&fixtures, "a", [("a", SuiAddress::ZERO), ("b", b_id)]).await?;
-        compile_package(a_src)
+        (
+            compile_package(a_src.clone()),
+            publish_package(context, sender, a_src).await,
+        )
     };
     let client = context.get_client().await?;
     let verifier = BytecodeSourceVerifier::new(client.read_api(), false);
+    let a_addr: SuiAddress = a_ref.0.into();
+
+    // Skip deps and root
+    verifier
+        .verify_package(&a_pkg.package, false, None)
+        .await
+        .unwrap();
+    sleep(Duration::from_millis(1000)).await;
```

Comment:
Talked about this offline -- we shouldn't need these `sleep`s really, especially from one call to `verify_package` to the next, it indicates something is amiss because we should be able to run these all in parallel.